### ISSUE TITLE
Cannot generate release notes for manually uploaded Azure artifacts

### ISSUE THREAD

**Issue by rfennell** [rfennell/AzurePipelines#1966]
### Discussed in https://github.com/rfennell/AzurePipelines/discussions/1960

Need to add a guard clause no no attempt is made to try to generate release notes for manually uploaded Azure Artifacts

**Comment by rfennell** [comment]
@adjurdjevic I have tried to repo this issue, and I can't get the failure you see. My process is

1. Add Universal package (in my case a simple zip containing some text files) to an Azure DevOps project scoped artifact feed using the Azure CLI

```
az artifacts universal publish --organization https://dev.azure.com/richardfennell --project github --scope project --feed pipelineartifacts --name manualupload --version 1.0.0 --path c:\tmp\manualupload.zip --description "manually uploaded"
```

2. Add a universal package to the same feed, but do this using an Azure DevOps pipeline to there is a release history
3. Run a Classic release that uses these two artifacts and generate release notes, I see the following

```
Getting all artifacts in the current release...
The artifact [_manualupload] is an Azure Artifacts, expanding build details
The artifact [_sample-pipeline-artifact] is an Azure Artifacts, expanding build details
Found 2 artifacts for current release
Getting all artifacts in the most recent successful release [Release-1]...
The artifact [_sample-pipeline-artifact] is an Azure Artifacts, expanding build details
Found 1 artifacts for most recent successful release
Looking at artifact [_manualupload]
Artifact type [Build]
Build Definition ID [undefined]
Build Number: [undefined]
Is Primary: [undefined]
Looking for the [_manualupload] in the most recent successful release [Release-1]
Looking at artifact [_sample-pipeline-artifact]
Artifact type [Build]
Build Definition ID [102]
Build Number: [20210201.5]
Is Primary: [true]
Looking for the [_sample-pipeline-artifact] in the most recent successful release [Release-1]
Found artifact [_sample-pipeline-artifact] with build number [20210201.5] in release [Release-1]
Build for artifact [_sample-pipeline-artifact] has not changed. Nothing to do
Getting test associated with the latest build [8884]
Adding the build [8884] and its associations to the unified results object
Detected 0 commits/changesets between the current build and the last successful one
Detected 0 workitems between the current build and the last successful one
Found 0 tests associated with the build [8884] adding any not already in the global test list to the list
Found 0 manual tests associated with the build [8884] adding any not already in the global test list to the list
```

So I guess my test differs from your usage in some way.

That said, having the guard method you propose in #1967 makes good sense. I will make sure that passes all my other tests

**Comment by rfennell** [comment]
Interesting, if I re-run my test pipeline, on the 2nd run it does report the same error you see, strange

**Comment by rfennell** [comment]
Using #1967 released as 4.18.4

**Comment by adjurdjevic** [comment]
Hi Richard, 
I can confirm that issue is solved on my side with this fix. Thank you for help.

Br
Aleksandar
